### PR TITLE
 Add host[:port] property in appengine-web.xml to tell build server app host url when deploy it on other GAE implementation such as AppScale.

### PR DESCRIPTION
**Pull request by wanddy**
Add host[:port] property in appengine-web.xml to tell build server app host url when deploy it on other GAE implementation such as AppScale.
Signed-off-by: wanddy <wanddy@163.com>

### CODE REVIEW

**Comment by halatmit** [comment]
Thanks!



== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty

**Comment by jisqyv** [comment]
The logic here looks wrong. If you set the default value of appengine.host to "localhost:8888" then it will use it, even in production (because you test with appengineHost.get()==""). Perhaps the default should be the empty string both in the code and in the appengine-web.xml file. A comment in appengine-web.xml can explain how to use it.

-Jeff

**Comment by wanddy** [comment]
I have reviewed my code and configuration,and test on Appscale and GAE,it works fine by setting appengine.host or remain blank.

File: appinventor/appengine/src/com/google/appinventor/server/project/youngandroid/YoungAndroidProjectService.java [hunk 1]
```diff
@@ -114,6 +114,9 @@
   // host[:port] to use for connecting to the build server
   private static final Flag<String> buildServerHost =
       Flag.createFlag("build.server.host", "localhost:9990");
+  // host[:port] to tell build server app host url
+  private static final Flag<String> appengineHost =
+      Flag.createFlag("appengine.host", "localhost:8080");
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/server/project/youngandroid/YoungAndroidProjectService.java]: @wanddy Perhaps the default here should be "" instead of localhost:8080. In theory this default will never get used because the appengine-web.xml file sets the property to "". This is why I was confused about whether or not the code would work.

-Jeff

P.S. Sorry for the delay in all of this. Been busy :-)

File: appinventor/buildserver/src/com/google/appinventor/buildserver/Compiler.java [hunk 2]
```diff
@@ -319,7 +320,7 @@ private boolean writeAndroidManifest(File manifestFile, Set<String> permissionsN
 
     // TODO(user): Use com.google.common.xml.XmlWriter
     try {
-      BufferedWriter out = new BufferedWriter(new FileWriter(manifestFile));
+      BufferedWriter out = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(manifestFile), "UTF-8"));
```
Comment (jisqyv) [inline appinventor/buildserver/src/com/google/appinventor/buildserver/Compiler.java]: @wanddy This change should probably be in its own PR so it winds up in its own commit.

**Comment by jisqyv** [comment]
@wanddy Sorry for the delay, LGTM. Currently squashed and submitted to Gerrit. Should be merged shortly.

**Comment by jisqyv** [comment]
@wanddy Note: I fixed some whitespace issues

**Comment by jisqyv** [comment]
MERGED
